coverage.py

```diff
@@ -179,52 +185,37 @@ def finalize_batch(self):
         pass
 
 
-class IdentifierBasedCoverageProvider(CoverageProvider):
-    """Run Identifiers from one DataSource (the input DataSource) through
-    code associated with another DataSource (the output
-    DataSource). If the code returns success, add a CoverageRecord for
-    the Identifier and the output DataSource, so that the record
-    doesn't get processed next time.
-    """
-
-    @property
-    def editions_that_need_coverage(self):
-        """Find all Identifiers lacking coverage from this CoverageProvider.
-
-        Identifiers are selected randomly to reduce the effect of
-        persistent errors.
-        """
-        return Identifier.missing_coverage_from(
-            self._db, self.input_sources, self.output_source).order_by(
-                func.random())
-
+class BibliographicCoverageProvider(CoverageProvider):
+    """Fill in bibliographic metadata for records.
 
-class BibliographicCoverageProvider(IdentifierBasedCoverageProvider):
-    """Fill in bibliographic metadata for records."""
+    Ensures that a given DataSource provides coverage for all
+    identifiers of the type primarily used to identify books from that
+    DataSource.
 
-    def __init__(self, _db, api, datasource):
+    e.g. ensures that we get Overdrive coverage for all Overdrive IDs.
+    """
+    def __init__(self, _db, api, datasource, workset_size=10):
         self._db = _db
         self.api = api
-        self.input_source = DataSource.lookup(_db, datasource)
-        self.output_source = DataSource.lookup(_db, datasource)
+        output_source = DataSource.lookup(_db, datasource)
+        input_identifier_types = [output_source.primary_identifier_type]
         service_name = "%s Bibliographic Monitor" % datasource
-        super(BibliographicCoverageProvider, self).__init__(service_name,
-                self.input_source, self.output_source)
+        super(BibliographicCoverageProvider, self).__init__(
+            service_name,
+            input_identifier_types, output_source,
+            workset_size=workset_size
+        )
 
     def process_batch(self):
         """Returns a list of successful identifiers and CoverageFailures"""
         raise NotImplementedError
 
-    def set_presentation_ready(self, identifier, metadata):
-        """Given a metadata object as content, finds an identifier's work
-        and sets it presentation ready.
-
-        Returns either a coverage failure or the processed identifier.
+    def work(self, identifier):
```

Comment:
I split set_presentation_ready into three methods: work(), set_metadata(), and set_presentation_ready(). This is because the metadata wrangler can't set a work as presentation-ready based on data from a single CoverageProvider.